### PR TITLE
always allow trying mosek_cli

### LINKED ISSUE #963: unexpected build script messages on windows
pointed out by @galbramc on #961. quoting him,

"
@bqpd I don't think you fixed the problem though. I'm still seeing the message:

```
# Looking for mosek_cli
# Build script does not support mosek_cli your architecture (64bit)
# Did not find mosek_cli
```
https://acdl.mit.edu/csi/job/gpkit_Push/buildnode=windows10x64,optimizer=mosek/356/console

If you look further down below it says:

```
Replaced found solvers (['mosek', 'cvxopt']) with environment var GPKITSOLVERS (mosek, mosek_cli)
```
"

### PR DESCRIPTION
**Pull request by bqpd**
Closes #963 
Addresses debugging flow of #959

### CODE REVIEW

**Comment by bqpd** [comment]
@whoburg, ready for review.

**Comment by bqpd** [comment]
Failing only because of MOSEK 8. @whoburg this is ready for review.

**Comment by galbramc** [comment]
Still not the right answer. You now find the 32 bit mosek_cli on WIndows 7, but gpkit will fail if it tries to use it. You don't see a failure at the moment because I'm forcing only the cvxopt solver.

**Comment by whoburg** [comment]
@bqpd, waiting to hear a response to @galbramc's concern above before reviewing.

Also, what is the status of Mosek 8 support?

**Comment by bqpd** [comment]
re: MOSEK 8, not yet begun. 
re: finding 32bit MOSEK, I'm a little confused why a 32bit MOSEK is in the 64bit Program Files folder on a 64bit operating system... For now, my priority is fixing observed problems from users; making sure that MOSEK 32bit has a friendly error is next, along with testing the other Windows install bugs.

**Comment by galbramc** [comment]
Oops, I had it backwards. Windows 7 has a 32 bit version of python using pythonxy, but mosek is 64bit. That combination does not work. I never upgraded windows 7 to us anaconda after we realized that you could install cvxopt with anaconda.

**Comment by bqpd** [comment]
oh hmm, so it's py32 sys.exec not working with mosek64. updated the issue accordingly...

**Comment by galbramc** [comment]
Yeah I don't know how the mosek_cli works, but from what I recall the mixture of bits don't play nice with each other. Let me turn of the forcing of cvxopt and see what happens.

**Comment by galbramc** [comment]
test this please

**Comment by galbramc** [comment]
Not the error that I recalled, but this is what you get now:

https://acdl.mit.edu/csi/job/gpkit_PullRequest_Unit/buildnode=windows7x64,optimizer=cvxopt/1744/console

**Comment by bqpd** [comment]
Hmm! It looks like it successfully ran the `mosek_cli` command? But did not like `mosek`, and so did not set the path...that's a bug. I'll push a fix up.

**Comment by bqpd** [comment]
Looks like it's using `mosek_cli` just fine!

**Comment by galbramc** [comment]
Apparently. I don't remember what the problem was, but ill update jenkins
to run mosek_cli on Windows 7.

**Comment by galbramc** [comment]
Test this please

File: gpkit/_mosek/cli_expopt.py [hunk 1]
```diff
@@ -35,8 +35,9 @@ def imize_fn(path=None, clearfiles=True):
     if not path:
         path = tempfile.mkdtemp()
     filename = path + os.sep + "gpkit_mosek"
-    os.environ['PATH'] = (os.environ['PATH'] + ':%s' %
-                          settings["mosek_bin_dir"])
+    if "mosek_bin_dir" in settings:
+        os.environ['PATH'] = (os.environ['PATH'] + ':%s' %
+                              settings["mosek_bin_dir"])
```
Comment (whoburg) [inline gpkit/_mosek/cli_expopt.py]: `":".join([os.environ['PATH'], settings["mosek_bin_dir"]])`

seems cleaner

File: gpkit/build.py [hunk 2]
```diff
@@ -121,7 +113,7 @@ class CVXopt(SolverBackend):
     name = "cvxopt"
 
     def look(self):
-        "Attempts to import mskexpopt."
+        "Attempts to import cvxopt."
```
Comment (whoburg) [inline gpkit/build.py]: good catch!

**Comment by bqpd** [comment]
@whoburg, may we merge?

File: gpkit/build.py [hunk 3]
```diff
@@ -279,7 +271,12 @@ def build(self):
 
 def build_gpkit():
     "Builds GPkit"
-    global settings  # pylint: disable=global-variable-not-assigned
+    try:
+        import gpkit
+        log("# Moving to the directory from which GPkit was imported.")
+        os.chdir(gpkit.__path__[0])
```
Comment (whoburg) [inline gpkit/build.py]: hmm, does this have the potential to cause problems if there are multiple installs of GPkit and the user wants to run the build script to overwrite the currently-installed one?

File: gpkit/nomials/array.py [hunk 4]
```diff
@@ -59,7 +59,7 @@ class NomialArray(np.ndarray):
     __repr__ = _repr
     _repr_latex_ = _repr_latex_
 
-    def str_without(self, excluded=()):
+    def str_without(self, excluded=None):
```
Comment (whoburg) [inline gpkit/nomials/array.py]: fine, but seems unrelated?

**Comment by bqpd** [comment]
oops, @galbramc, I think it's still finding MOSEK 8

**Comment by bqpd** [comment]
@whoburg the code is ready for review, though tests are pending.

**Comment by bqpd** [comment]
retest this please

File: gpkit/nomials/array.py (hunk 4 above)
Comment (bqpd) [inline gpkit/nomials/array.py]: heh, yeah.

**Comment by galbramc** [comment]
Hold on, I'm updating the testing script.

**Comment by galbramc** [comment]
test this please

File: gpkit/_mosek/cli_expopt.py (hunk 1 above)
Comment (bqpd) [inline gpkit/_mosek/cli_expopt.py]: I like it!

Comment (bqpd) [inline gpkit/_mosek/cli_expopt.py]: Still doesn't fit on one line, sadly.

File: gpkit/build.py (hunk 3 above)
Comment (bqpd) [inline gpkit/build.py]: I believe that the install step will remove the already installed one before building, and this passes tests and helps with #959 so I figured it was worth a try.

**Comment by whoburg** [comment]
code is approved but I don't want us to merge until two conditions are true:

1) all three jobs on the github home page are showing green
2) all tests pass on this PR

**Comment by bqpd** [comment]
retest this please

**Comment by bqpd** [comment]
(@galbramc, on unit tests but not dependencies still finding mosek8 on win10)

**Comment by galbramc** [comment]
I just reinstalled mosek 7.

**Comment by galbramc** [comment]
test this please

**Comment by whoburg** [comment]
@bqpd, looks good to merge.

**Comment by bqpd** [comment]
It is done!
